File: Modules/Common/Resources/SIE.Web.Resources/CalendarSchemes/Commands/ShowHolidayCommand.js
SIE.defineCommand('SIE.Web.Resources.CalendarSchemes.Commands.ShowHolidayCommand', {
    meta: { text: "查看法定假期", group: "edit", iconCls: "icon-FileEye icon-blue" },
    execute: function (view, source) {
        CRT.Workbench.addPage({
            entityType: 'SIE.Resources.Holidays.Holiday,SIE.Resources',
            title: '法定假期'.L10N(),
            module: 'SIE.Resources.Holidays.Holiday,SIE.Resources',//view.module,
            ignoreQuery: false,
            isAggt: true
        });
    }
});

File: Modules/Common/Resources/SIE.Wpf.Resources/CalendarSchemes/Commands/ShowHolidayCommand.cs
using SIE.Resources.Holidays;
using SIE.Wpf.Command;
using System;

namespace SIE.Wpf.Resources.CalendarSchemes.Commands
{
    /// <summary>
    /// 查看法定假期
    /// </summary>
    [Command(ImageName = "FileEye", Label = "查看法定假期", GroupType = 20)]
    public class ShowHolidayCommand : ClientCommand
    {
        /// <summary>
        /// 查看日历方案
        /// </summary>
        protected override void ExecuteCore()
        {
            string key = CRT.Workbench.CreateKey(ViewConfig.ListView, typeof(Holiday), null);

            CRT.Workbench.ShowView(key, v =>
            {
                v.Title = Meta.Label.L10N();
                var ui = new ListUITemplate(typeof(Holiday)).CreateUI();
                var query = ui.MainView.QueryView;
                if (query != null)
                {
                    query.TryExecuteQuery();
                }

                v.Closing += (o, e) =>
                {
                    if (ui.MainView.Data.IsDirty)
                    {
                        if (CRT.MessageService.AskQuestion("数据未保存，确定退出吗".L10N()))
                        {
                            e.Cancel = false;
                        }
                        else
                        {
                            e.Cancel = true;
                        }
                    }
                };
                return ui;
            });
        }
    }
}
